net: support for easier out of tree net stack implementations

jhirsi opened the pull request:
Current default functionality is not changed, but this PR adds customization points by introducing 2 new Kconfig items:
- NET_LINK_ADDR_CUSTOM_LENGTH that allows to set custom link layer address length if your link layer technology is not supported directly
- NET_MGMT_EVENT_INFO_DEFAULT_DATA_SIZE is used to set the default size of the data field in the net_mgmt_event_info structure. This change allows the user to configure the size of the data field according to their needs.

jhirsi committed:
net: possibility to set custom link layer address length

This commit adds a new Kconfig option NET_LINK_ADDR_CUSTOM_LENGTH
that allows to set custom link layer address length if your link layer
technology is not supported directly. If this option is set to a value
greater than 0, that value is used as link layer address length.

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

maass-hamburg commented on subsys/net/ip/Kconfig.mgmt:
```diff
@@ -95,6 +95,14 @@ config NET_MGMT_EVENT_INFO
+	default 32
+	help
+	  The default size of the data which can be passed along with an event.
+endif
```
extend the if to after `config NET_MGMT_EVENT_MONITOR` so `NET_MGMT_EVENT_INFO` can be removed from its depends on

jhirsi committed:
net: ip: new Kconfig NET_MGMT_EVENT_INFO_DEFAULT_DATA_SIZE

The NET_MGMT_EVENT_INFO_DEFAULT_DATA_SIZE is used to set the default
size of the data field in the net_mgmt_event_info structure. This
change allows the user to configure the size of the data field
according to their needs.

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>
